### ISSUE TITLE
Errors after leaving combat from a dungeon boss encounter in LFG

### ISSUE THREAD

**Issue by frozn** [frozn/TipTac#132]
Source: https://www.curseforge.com/wow/addons/tiptac-reborn?comment=21

**Hellsy55:**
As of today's Retail update (10.0.2) got these errors after leaving combat from a dungeon boss encounter in LFG:

```
3x Interface/AddOns/TipTac/ttCore.lua:822: calling 'GetObjectType' on bad self (Usage: local objectType = self:GetObjectType())
[string "=[C]"]: in function `GetObjectType'
[string "@Interface/AddOns/TipTac/ttCore.lua"]:822: in function `SetPadding'
[string "@Interface/AddOns/TipTac/ttCore.lua"]:1333: in function <Interface/AddOns/TipTac/ttCore.lua:1331>
[string "=[C]"]: in function `Show'
[string "@Interface/SharedXML/Tooltip/TooltipDataHandler.lua"]:284: in function <Interface/SharedXML/Tooltip/TooltipDataHandler.lua:240>
[string "=[C]"]: in function `securecallfunction'
[string "@Interface/SharedXML/Tooltip/TooltipDataHandler.lua"]:237: in function <Interface/SharedXML/Tooltip/TooltipDataHandler.lua:236>
[string "=(tail call)"]: ?
[string "@Interface/AddOns/Blizzard_NamePlates/Blizzard_NamePlates.lua"]:860: in function <...e/AddOns/Blizzard_NamePlates/Blizzard_NamePlates.lua:853>
[string "=[C]"]: ?

Locals:
Skipped (In Encounter)
```

```
1x Interface/AddOns/TipTac/ttCore.lua:1750: calling 'GetParent' on bad self (Usage: local parent = self:GetParent())
[string "=[C]"]: in function `GetParent'
[string "@Interface/AddOns/TipTac/ttCore.lua"]:1750: in function `SetCenterColor'
[string "@Interface/SharedXML/SharedTooltipTemplates.lua"]:52: in function <Interface/SharedXML/SharedTooltipTemplates.lua:41>
[string "=[C]"]: in function `SharedTooltip_SetBackdropStyle'
[string "@Interface/FrameXML/GameTooltip.lua"]:449: in function <Interface/FrameXML/GameTooltip.lua:446>
[string "=[C]"]: ?
[string "=[C]"]: in function `Hide'
[string "@Interface/AddOns/Blizzard_NamePlates/Blizzard_NamePlates.lua"]:867: in function <...e/AddOns/Blizzard_NamePlates/Blizzard_NamePlates.lua:866>
[string "=[C]"]: in function `Hide'
[string "@Interface/SharedXML/Pools.lua"]:162: in function `resetterFunc'
[string "@Interface/SharedXML/Pools.lua"]:65: in function `Release'
[string "@Interface/SharedXML/Pools.lua"]:76: in function `ReleaseAll'
[string "@Interface/AddOns/Blizzard_NamePlates/Blizzard_NamePlates.lua"]:764: in function `UpdateBuffs'
[string "@Interface/AddOns/Blizzard_NamePlates/Blizzard_NamePlates.lua"]:253: in function `OnUnitAuraUpdate'
[string "@Interface/AddOns/Blizzard_NamePlates/Blizzard_NamePlates.lua"]:89: in function <...e/AddOns/Blizzard_NamePlates/Blizzard_NamePlates.lua:69>

Locals:
Skipped (In Encounter)
```

**Comment by frozn** [comment]
Can't reproduce this. Asked him, if he use a special nameplate addon.

**Comment by frozn** [comment]
**Answer from Hellsy55:**
Only using plater. Had no errors on their end.

**Comment by lowpita** [comment]
Hi. I got these similar errors while in raid boss encounters: 

6x Interface/AddOns/TipTac/ttCore.lua:1790: calling 'GetParent' on bad self (Usage: local parent = self:GetParent())
[string "=[C]"]: in function `GetParent'
[string "@Interface/AddOns/TipTac/ttCore.lua"]:1790: in function `SetCenterColor'
[string "@Interface/SharedXML/SharedTooltipTemplates.lua"]:52: in function <Interface/SharedXML/SharedTooltipTemplates.lua:41>
[string "=[C]"]: in function `SharedTooltip_SetBackdropStyle'
[string "@Interface/FrameXML/GameTooltip.lua"]:449: in function <Interface/FrameXML/GameTooltip.lua:446>
[string "=[C]"]: ?
[string "=[C]"]: ?
[string "=[C]"]: in function `Hide'
[string "@Interface/AddOns/Blizzard_UIWidgets/Blizzard_UIWidgetTemplateBase.lua"]:109: in function `OnLeave'
[string "@Interface/AddOns/Blizzard_UIWidgets/Blizzard_UIWidgetTemplateBase.lua"]:626: in function <...Blizzard_UIWidgets/Blizzard_UIWidgetTemplateBase.lua:625>

Locals:
(*temporary) = Frame {
 BottomLeftCorner = Texture {
 }
 OnBackdropLoaded = <function> defined @Interface/SharedXML/Backdrop.lua:152
 GetBackdropColor = <function> defined @Interface/SharedXML/Backdrop.lua:390
 SetBackdropBorderColor = <function> defined @Interface/AddOns/TipTac/ttCore.lua:1781
 RightEdge = Texture {
 }
 BottomEdge = Texture {
 }
 SetBackdrop = <function> defined @Interface/AddOns/TipTac/ttCore.lua:1765
 GetCenterColor = <function> defined @Interface/SharedXML/NineSlice.lua:255
 GetBackdropBorderColor = <function> defined @Interface/SharedXML/Backdrop.lua:409
 GetBackdrop = <function> defined @Interface/SharedXML/Backdrop.lua:347
 SetBorderBlendMode = <function> defined @Interface/SharedXML/Backdrop.lua:266
 SetCenterColor = <function> defined @Interface/AddOns/TipTac/ttCore.lua:1789
 GetFrameLayoutTextureKit = <function> defined @Interface/SharedXML/NineSlice.lua:236
 SetVertexColor = <function> defined @Interface/SharedXML/NineSlice.lua:285
 SetupTextureCoordinates = <function> defined @Interface/SharedXML/Backdrop.lua:214
 GetBorderColor = <function> defined @Interface/SharedXML/NineSlice.lua:273
 TopRightCorner = Texture {
 }
 SetupPieceVisuals = <function> defined @Interface/SharedXML/Backdrop.lua:246
 GetEdgeSize = <function> defined @Interface/SharedXML/Backdrop.lua:188
 0 = <userdata>
 OnLoad = <function> defined @Interface/SharedXML/NineSlice.lua:241
 TopEdge = Texture {
 }
 ApplyBackdrop = <function> defined @Interface/AddOns/TipTac/ttCore.lua:1757
 HasBackdropInfo = <function> defined @Interface/SharedXML/Backdrop.lua:278
 OnBackdropSizeChanged = <function> defined @Interface/SharedXML/Backdrop.lua:182
 Center = Texture {
 }
 TopLeftCorner = Texture {
 }
 ClearBackdrop = <function> defined @Interface/SharedXML/Backdrop.lua:282
 GetFrameLayoutType = <function> defined @Interface/SharedXML/NineSlice.lua:232
 BottomRightCorner = Texture {
 }
 SetBackdropColor = <function> defined @Interface/AddOns/TipTac/ttCore.lua:1773
 LeftEdge = Texture {
 }
 SetBorderColor = <function> defined @Interface/AddOns/TipTac/ttCore.lua:1797
}

12x Interface/AddOns/TipTac/ttCore.lua:823: calling 'GetObjectType' on bad self (Usage: local objectType = self:GetObjectType())
[string "=[C]"]: in function `GetObjectType'
[string "@Interface/AddOns/TipTac/ttCore.lua"]:823: in function `SetPadding'
[string "@Interface/AddOns/TipTac/ttCore.lua"]:1372: in function <Interface/AddOns/TipTac/ttCore.lua:1365>
[string "=[C]"]: ?
[string "=[C]"]: in function `SetShown'
[string "@Interface/AddOns/Blizzard_UIWidgets/Blizzard_UIWidgetTemplateBase.lua"]:102: in function `OnEnter'
[string "@Interface/AddOns/Blizzard_UIWidgets/Blizzard_UIWidgetTemplateBase.lua"]:621: in function <...Blizzard_UIWidgets/Blizzard_UIWidgetTemplateBase.lua:620>

Locals:
(*temporary) = EmbeddedItemTooltip {
 0 = <userdata>
 SetQuestLogRewardSpell = <function> defined @Interface/SharedXML/Tooltip/TooltipDataHandler.lua:444
 ItemTooltip = Frame {
 }
 SetHyperlink = <function> defined =[C]:-1
 SetAzeriteEssence = <function> defined =[C]:-1
 SetVoidDepositItem = <function> defined @Interface/SharedXML/Tooltip/TooltipDataHandler.lua:444
 SetTraitEntry = <function> defined @Interface/SharedXML/Tooltip/TooltipDataHandler.lua:444
 ClearBackdrop = <function> defined @Interface/SharedXML/Backdrop.lua:282
 SetLootCurrency = <function> defined @Interface/SharedXML/Tooltip/TooltipDataHandler.lua:444
 SetWorldCursor = <function> defined @Interface/FrameXML/GameTooltip.lua:952
 SetLootItem = <function> defined @Interface/SharedXML/Tooltip/TooltipDataHandler.lua:444
 GetSpell = <function> defined @Interface/FrameXML/GameTooltip.lua:990
 ttGradient = Texture {
 }
 SetQuestLogSpecialItem = <function> defined @Interface/SharedXML/Tooltip/TooltipDataHandler.lua:444
 SetRecipeReagentItem = <function> defined =[C]:-1
 SetMerchantItem = <function> defined @Interface/SharedXML/Tooltip/TooltipDataHandler.lua:444
 BottomOverlay = Texture {
 }
 SetSocketedRelic = <function> defined @Interface/SharedXML/Tooltip/TooltipDataHandler.lua:444
 ProcessInfo = <function> defined @Interface/SharedXML/Tooltip/TooltipDataHandler.lua:236
 NineSlice = Frame {
 }
 OnLoad = <function> defined @Interface/FrameXML/GameTooltip.lua:927
 TopEdge = Texture {
 }
 SetUnitDebuffByAuraInstanceID = <function> defined =[C]:-1
 SetBagItem = <function> defined @Interface/SharedXML/Tooltip/TooltipDataHandler.lua:444
 SetBackdropColor = <function> defined @Interface/AddOns/TipTac/ttCore.lua:1732
 LeftEdge = Texture {
 }
 AddLineDataText = <function> defined @Interface/SharedXML/Tooltip/TooltipDataHandler.lua:324
 SetSpellBookItem = <function> defined =[C]:-1
 textLeft1Font = "GameTooltipHeaderText"
 SetShapeshift = <function> defined @Interface/SharedXML/Tooltip/TooltipDataHandler.lua:444
 SetArtifactPowerByID = <function> defined @Interface/SharedXML/Tooltip/TooltipDataHandler.lua:444
 textRight2Font = "GameTooltipText"
 TopOverlay = Texture {
 }
 SetSocketedItem = <function> defined @Interface/SharedXML/Tooltip/TooltipDataHandler.lua:444
 SetUnitDebuff = <function> defined =[C]:-1
 SetMountBySpellID = <function> defined @Interface/SharedXML/Tooltip/TooltipDataHandler.lua:444
 SetBorderBlendMode = <function> defined @Interface/SharedXML/Backdrop.lua:266
 SetToyByItemID = <function> defined =[C]:-1
 TextLeft1 = EmbeddedItemTooltipTextLeft1 {
 }
 SetAzeritePower = <function> defined @Interface/SharedXML/Tooltip/TooltipDataHandler.lua:444
 SetLFGDungeonShortageReward = <function> defined =[C]:-1
 OnBackdropSizeChanged = <function> defined @Interface/SharedXML/Backdrop.lua:182
 SetArtifactItem = <function> defined @Interface/SharedXML/Tooltip/TooltipDataHandler.lua:444
 StatusBar = EmbeddedItemTooltipStatusBar {
 }
 SetPossession = <function> defined @Interface/SharedXML/Tooltip/TooltipDataHandler.lua:444
 SetUnitAura = <function> defined =[C]:-1
 SetUnitBuffByAuraInstanceID = <function> defined =[C]:-1
 SetConduit = <function> defined =[C]:-1
 SetupPieceVisuals = <function> defined @Interface/SharedXML/Backdrop.lua:246
 SetMinimapMouseover = <function> defined @Interface/SharedXML/Tooltip/TooltipDataHandler.lua:444
 GetItem = <function> defined @Interface/FrameXML/GameTooltip.lua:986
 SetRecipeResultItem = <function> defined @Interface/SharedXML/Tooltip/TooltipDataHandler.lua:444
 SetQuestPartyProgress = <function> defined =[C]:-1
 ProcessLineData = <function> defined @Interface/SharedXML/Tooltip/TooltipDataHandler.lua:303
 OnBackdropLoaded = <function> defined @Interface/SharedXML/Backdrop.lua:152
 SetPvpTalent = <function> defined @Interface/SharedXML/Tooltip/TooltipDataHandler.lua:444
 SetExistingSocketGem = <function> defined @Interface/SharedXML/Tooltip/TooltipDataHandler.lua:444
 SetUnit = <function> defined =[C]:-1
 SetTotem = <function> defined @Interface/SharedXML/Tooltip/TooltipDataHandler.lua:444
 SetVoidWithdrawalItem = <function> def
 
 This one was out of combat: 
 
 4x bad argument #1 to '?' (Usage: self:SetVertexColor(color [, a]))
[string "=[C]"]: in function `SetVertexColor'
[string "@Interface/AddOns/TipTac/ttCore.lua"]:659: in function <Interface/AddOns/TipTac/ttCore.lua:640>
[string "@Interface/AddOns/TipTac/ttCore.lua"]:735: in function `ApplySettings'
[string "@Interface/AddOns/TipTac/ttCore.lua"]:2145: in function `AddModifiedTip'
[string "@Interface/AddOns/TipTac/ttCore.lua"]:2015: in function `Acquire'
[string "@Interface/AddOns/SavedInstances/Core/Core.lua"]:3479: in function `ShowTooltip'
[string "@Interface/AddOns/SavedInstances/Core/Core.lua"]:2590: in function `OnEnter'
[string "@Interface/AddOns/Masque/Libs/LibDBIcon-1.0-45/LibDBIcon-1.0.lua"]:77: in function <...e/AddOns/Masque/Libs/LibDBIcon-1.0/LibDBIcon-1.0.lua:60>

Locals:
(*temporary) = "bad argument #1 to '?' (Usage: self:SetVertexColor(color [, a]))"

**Comment by lowpita** [comment]
Hi. Today I got this error too which seems to be related to all of the above but the savedinstances/masque one:

14x TipTacItemRef/ttItemRef.lua:2359: Attempt to access forbidden object from code tainted by an AddOn
[string "=[C]"]: in function `AddLine'
[string "@TipTacItemRef/ttItemRef.lua"]:2359: in function `spell'
[string "@TipTacItemRef/ttItemRef.lua"]:592: in function <TipTacItemRef/ttItemRef.lua:581>
[string "=[C]"]: in function `setFunction'
[string "@Blizzard_NamePlates/Blizzard_NamePlates.lua"]:860: in function <...eBlizzard_NamePlates/Blizzard_NamePlates.lua:853>
[string "=[C]"]: ?

Locals:
(*temporary) = NamePlateTooltip {
 0 = <userdata>
 SetQuestLogRewardSpell = <function> defined @SharedXML/Tooltip/TooltipDataHandler.lua:444
 GetBackdropColor = <function> defined @SharedXML/Backdrop.lua:390
 SetHyperlink = <function> defined =[C]:-1
 SetAzeriteEssence = <function> defined =[C]:-1
 SetVoidDepositItem = <function> defined @SharedXML/Tooltip/TooltipDataHandler.lua:444
 SetTraitEntry = <function> defined @SharedXML/Tooltip/TooltipDataHandler.lua:444
 ClearBackdrop = <function> defined @SharedXML/Backdrop.lua:282
 SetLootCurrency = <function> defined @SharedXML/Tooltip/TooltipDataHandler.lua:444
 SetWorldCursor = <function> defined @FrameXML/GameTooltip.lua:952
 SetLootItem = <function> defined @SharedXML/Tooltip/TooltipDataHandler.lua:444
 GetSpell = <function> defined @FrameXML/GameTooltip.lua:990
 ttGradient = Texture {
 }
 SetQuestLogSpecialItem = <function> defined @SharedXML/Tooltip/TooltipDataHandler.lua:444
 SetRecipeReagentItem = <function> defined =[C]:-1
 SetMerchantItem = <function> defined @SharedXML/Tooltip/TooltipDataHandler.lua:444
 BottomOverlay = Texture {
 }
 SetSocketedRelic = <function> defined @SharedXML/Tooltip/TooltipDataHandler.lua:444
 ProcessInfo = <function> defined @SharedXML/Tooltip/TooltipDataHandler.lua:236
 NineSlice = Frame {
 }
 OnLoad = <function> defined @FrameXML/GameTooltip.lua:927
 TopEdge = Texture {
 }
 SetUnitDebuffByAuraInstanceID = <function> defined =[C]:-1
 SetBagItem = <function> defined @SharedXML/Tooltip/TooltipDataHandler.lua:444
 SetBackdropColor = <function> defined @TipTac/ttCore.lua:1746
 LeftEdge = Texture {
 }
 AddLineDataText = <function> defined @SharedXML/Tooltip/TooltipDataHandler.lua:324
 SetSpellBookItem = <function> defined =[C]:-1
 textLeft1Font = "GameTooltipHeaderText"
 SetShapeshift = <function> defined @SharedXML/Tooltip/TooltipDataHandler.lua:444
 SetArtifactPowerByID = <function> defined @SharedXML/Tooltip/TooltipDataHandler.lua:444
 textRight2Font = "GameTooltipText"
 TopOverlay = Texture {
 }
 SetSocketedItem = <function> defined @SharedXML/Tooltip/TooltipDataHandler.lua:444
 SetUnitDebuff = <function> defined =[C]:-1
 SetMountBySpellID = <function> defined @SharedXML/Tooltip/TooltipDataHandler.lua:444
 SetBorderBlendMode = <function> defined @SharedXML/Backdrop.lua:266
 SetTrainerService = <function> defined @SharedXML/Tooltip/TooltipDataHandler.lua:444
 TextLeft1 = NamePlateTooltipTextLeft1 {
 }
 SetAzeritePower = <function> defined @SharedXML/Tooltip/TooltipDataHandler.lua:444
 SetLFGDungeonShortageReward = <function> defined =[C]:-1
 OnBackdropSizeChanged = <function> defined @SharedXML/Backdrop.lua:182
 SetArtifactItem = <function> defined @SharedXML/Tooltip/TooltipDataHandler.lua:444
 StatusBar = NamePlateTooltipStatusBar {
 }
 SetPossession = <function> defined @SharedXML/Tooltip/TooltipDataHandler.lua:444
 SetUnitAura = <function> defined =[C]:-1
 SetUnitBuffByAuraInstanceID = <function> defined =[C]:-1
 SetConduit = <function> defined =[C]:-1
 SetupPieceVisuals = <function> defined @SharedXML/Backdrop.lua:246
 SetMinimapMouseover = <function> defined @SharedXML/Tooltip/TooltipDataHandler.lua:444
 GetItem = <function> defined @FrameXML/GameTooltip.lua:986
 SetRecipeResultItem = <function> defined @SharedXML/Tooltip/TooltipDataHandler.lua:444
 SetQuestPartyProgress = <function> defined =[C]:-1
 ProcessLineData = <function> defined @SharedXML/Tooltip/TooltipDataHandler.lua:303
 OnBackdropLoaded = <function> defined @SharedXML/Backdrop.lua:152
 SetPvpTalent = <function> defined @SharedXML/Tooltip/TooltipDataHandler.lua:444
 SetExistingSocketGem = <function> defined @SharedXML/Tooltip/TooltipDataHandler.lua:444
 SetUnit = <function> defined =[C]:-1
 SetTotem = <function> defined @I
 
I tried reproducing them, and I think what triggers it is mouseovering an enemy nameplate debuff which turns into a friendly/neutral one before/after combat ends, like the last boss from prepatch world event, Uldaman: Legacy of Tyr, while also having plater on. Got all three errors, but the masque/savedinstances on, at once.  I don't know if it happens without plater, but I cant reproduce the error without it because my default nameplates won't show up If i don't use the addon.

**Comment by lowpita** [comment]
Recorded a clip reproducing it: https://medal.tv/es/games/world-of-warcraft/clips/Jg0U0H47Ee6zj/d1337P41oOqh

**Comment by frozn** [comment]
Should be fixed with latest release [v23.01.23](https://github.com/frozn/TipTac/releases/tag/v23.01.23-release). Please try again with this version.

**Comment by lowpita** [comment]
Hey. I didn't have this issue happening before 23.01.23 because it seems plater devs (or blizzard) changed something to instantly hide the debuff widget when the npc turned friendly/neutral from hostile, so the debuffs won't show up anymore after it for reproducing this mouseover thing.
